test/e2e/sched/sched_test.go

```diff
@@ -68,6 +72,88 @@ var _ = Describe("[Scheduler] imageReplacement", func() {
 
 				return deploy.Spec.Template.Spec.Containers[0].Image == e2eimages.SchedTestImageCI
 			}, time.Minute, time.Second*10).Should(BeTrue())
+
+			By("reverting NROS changes")
+			err = e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), nroSchedObj)
+			Expect(err).ToNot(HaveOccurred())
+
+			nroSchedObj.Spec = objects.TestNROScheduler().Spec
+
+			err = e2eclient.Client.Update(context.TODO(), nroSchedObj)
+			Expect(err).ToNot(HaveOccurred())
+
+			// find deployment by the ownerReference
+			dp, err := schedutils.GetDeploymentByOwnerReference(nroSchedObj.GetUID())
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(wait.ForDeploymentComplete(e2eclient.Client, dp, time.Second*30, time.Minute*2)).ToNot(HaveOccurred())
+		})
+
+		It("should react to owned objects changes", func() {
+			var err error
+			nroSchedObj := objects.TestNROScheduler()
+
+			err = e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), nroSchedObj)
+			Expect(err).ToNot(HaveOccurred())
+
+			cmList := &corev1.ConfigMapList{}
+			err = e2eclient.Client.List(context.TODO(), cmList)
+			Expect(err).ToNot(HaveOccurred())
+
+			var nroCM *corev1.ConfigMap
+			for i := 0; i < len(cmList.Items); i++ {
+				if e2eobjects.IsOwnedBy(cmList.Items[i].ObjectMeta, nroSchedObj.ObjectMeta) {
+					nroCM = &cmList.Items[i]
+				}
+			}
+
+			initialCM := nroCM.DeepCopy()
+			nroCM.Data["somekey"] = "somevalue"
+
+			err = e2eclient.Client.Update(context.TODO(), nroCM)
```

Comment:
update can fail on a live cluster. We either need to use something smarter (patch?) or wrap the (get+)update in a loop

Reply:
Why that update would failed?

Reply:
I don't know the exact reason, but this is something we may happen any time in a cluster (part of how kube works) so robust code should tolerate it

Reply:
It's a bit worrying that a global wide-used function can fail without any good reason.
Does it mean we should stop using it? what about all the other places and production code we have which contain this function?

Of course, I can wrap it in a loop. But the general statement you raised here makes me question.

Reply:
Well this is kubernetes operates since day one pretty much. Retrying loop (which is what ultimately a control loop is about) and optimistic concurrency are the cornerstones of the platform. I'm not sure I understand the concern :\